rstudio/rstudio
# Electron font selection

### Intent
Part one to address #10438 and #10463 for font selection and seeing the font in the preview. Setting the font doesn't actually change the editor's font yet since it doesn't write or read a persisted preference.

### Approach
Changes the preload so it sends an event to the main process. On startup, the main process will get all the system fonts. On receiving the event, it can then return the list of fonts or the default monospace/proportional font. It's a synchronous event call from the renderer to the main process since the existing GWT code does not expect a promise. Given that the font list is generated on startup, the call to retrieve the fonts is retrieving the cache.

I did notice that the old code would fallback to Qt for querying system fonts. There is no fallback here if `node-system-fonts` did not discover any fonts.

### Automated Tests
None.

### QA Notes
There should be no duplicate fonts in the list and all font options should change the preview. The editor does not apply the font selection at this time. 

### Checklist
